front/src/modules/scenario/components/ScenarioExplorer/ScenarioExplorer.tsx

```diff
@@ -140,22 +161,31 @@ const ScenarioExplorer = ({
               <img src={scenarioIcon} alt="scenario icon" />
               <span className="scenario-explorator-card-head-legend">{t('scenarioLegend')}</span>
               <div className="scenario-explorator-card-head-scenario">
-                <span className="text-truncate" title={scenarioDetails.name}>
-                  {scenarioDetails.name}
+                <span className="text-truncate" title={scenarioDetails?.name}>
+                  {trainScheduleV2Activated ? scenarioV2Details?.name : scenarioDetails?.name}
                 </span>
-                {timetable && (
+                {trainScheduleV2Activated && timetableV2 ? (
```

Comment:
You already check if timetableV2 is defined l169
```suggestion
                {trainScheduleV2Activated ? (
```